### ISSUE TITLE
Error when creating new documents

### ISSUE THREAD

**Issue by kleener** [jgraph/drawio-nextcloud#16]
As soon as I create a new whiteboard or diagram with version 2.0.1, I get an error and I can't access any other files either. It seems the page is not loaded correctly anymore.
According to the log it seems to be related to the previews.
Version: 8.1.2

Error message 1:
[webdav] Fehler: Error: Call to undefined method OCA\Drawio\AppConfig::getPreviews() at <<closure>>

 0. /var/www/nextcloud/lib/private/Preview/ProviderV1Adapter.php line 48
    OCA\Drawio\Preview\DrawioPreview->isAvailable()
 1. /var/www/nextcloud/lib/private/PreviewManager.php line 257
    OC\Preview\ProviderV1Adapter->isAvailable()
 2. /var/www/nextcloud/apps/dav/lib/Connector/Sabre/FilesPlugin.php line 353
    OC\PreviewManager->isAvailable()
 3. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/PropFind.php line 95
    OCA\DAV\Connector\Sabre\FilesPlugin->OCA\DAV\Connector\Sabre\{closure}("*** sensitive parameters replaced ***")
 4. /var/www/nextcloud/apps/dav/lib/Connector/Sabre/FilesPlugin.php line 354
    Sabre\DAV\PropFind->handle()
 5. /var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    OCA\DAV\Connector\Sabre\FilesPlugin->handleGetProperties()
 6. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 1052
    Sabre\DAV\Server->emit()
 7. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 984
    Sabre\DAV\Server->getPropertiesByNode()
 8. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 1662
    Sabre\DAV\Server->getPropertiesIteratorForPath()
 9. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 1647
    Sabre\DAV\Server->writeMultiStatus()
10. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 346
    Sabre\DAV\Server->generateMultiStatus()
11. /var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpPropFind()
12. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 472
    Sabre\DAV\Server->emit()
13. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 253
    Sabre\DAV\Server->invokeMethod()
14. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 321
    Sabre\DAV\Server->start()
15. /var/www/nextcloud/apps/dav/lib/Server.php line 360
    Sabre\DAV\Server->exec()
16. /var/www/nextcloud/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
17. /var/www/nextcloud/remote.php line 171
    require_once("/var/www/cloud. ... p")

Error message 2:
[index] Fehler: Exception: Call to undefined method OCA\Drawio\AppConfig::getPreviews() in file '/var/www/nextcloud/apps/drawio/lib/Preview/DrawioPreview.php' line 68 at <<closure>>

0. /var/www/nextcloud/lib/private/AppFramework/App.php line 172
   OC\AppFramework\Http\Dispatcher->dispatch()
1. /var/www/nextcloud/lib/private/Route/Router.php line 298
   OC\AppFramework\App::main()
2. /var/www/nextcloud/lib/base.php line 1047
   OC\Route\Router->match()
3. /var/www/nextcloud/index.php line 36
   OC::handleRequest()

**Comment by marcatsmci** [comment]
Same problem: just try to enter the directory and the message **This directory is unavailable, please check the logs or contact the administrator** appears and the directory is inaccessible.

Deactivating the application allows you to see the contents of the directory again.

Debian 11.6 / PHP 8.1 / Nextcloud Hub 3 (25.0.2)  / Draw.io 2.0.1

**Comment by m-mohamedin** [comment]
Fixed in v2.0.2
Thanks for the report

**Comment by marcatsmci** [comment]
Version 2.0.2 does not fix the problem !

```

[webdav] Error: Error: Call to undefined method OCA\Drawio\AppConfig::getPreviews() at <<closure>>

 0. /var/www/nextcloud/lib/private/Preview/ProviderV1Adapter.php line 48
    OCA\Drawio\Preview\DrawioPreview->isAvailable()
 1. /var/www/nextcloud/lib/private/PreviewManager.php line 257
    OC\Preview\ProviderV1Adapter->isAvailable()
 2. /var/www/nextcloud/apps/dav/lib/Connector/Sabre/FilesPlugin.php line 353
    OC\PreviewManager->isAvailable()
 3. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/PropFind.php line 95
    OCA\DAV\Connector\Sabre\FilesPlugin->OCA\DAV\Connector\Sabre\{closure}("*** sensitive parameters replaced ***")
 4. /var/www/nextcloud/apps/dav/lib/Connector/Sabre/FilesPlugin.php line 354
    Sabre\DAV\PropFind->handle()
 5. /var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    OCA\DAV\Connector\Sabre\FilesPlugin->handleGetProperties()
 6. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 1052
    Sabre\DAV\Server->emit()
 7. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 984
    Sabre\DAV\Server->getPropertiesByNode()
 8. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 1662
    Sabre\DAV\Server->getPropertiesIteratorForPath()
 9. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 1647
    Sabre\DAV\Server->writeMultiStatus()
10. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 346
    Sabre\DAV\Server->generateMultiStatus()
11. /var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpPropFind()
12. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 472
    Sabre\DAV\Server->emit()
13. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 253
    Sabre\DAV\Server->invokeMethod()
14. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 321
    Sabre\DAV\Server->start()
15. /var/www/nextcloud/apps/dav/lib/Server.php line 360
    Sabre\DAV\Server->exec()
16. /var/www/nextcloud/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
17. /var/www/nextcloud/remote.php line 171
    require_once("/var/www/nextcl ... p")

PROPFIND /remote.php/dav/files/admin/Infra-Sch%C3%A9mas
from 194.230.158.76 by admin at 2023-01-05T08:19:10+00:00


```
